variations.py

```diff
@@ -1,14 +1,49 @@
 from copy import deepcopy
 
 from .booking import Unit
+from .booking import dataset_from_crownoutput
 from .booking import dataset_from_artusoutput
 from .utils import Selection
 from .utils import Variation
 
 import logging
 logger = logging.getLogger(__name__)
 
+class ChangeDatasetCrown(Variation):
+    """
+    Variation that with the method create makes a deepcopy of
+    the selections and actions inside the unit passed as argument and substitutes
+    the directory attribute with the according variation.
+
+    Args:
+        name (str): name used to identify the instance of
+            this class
+        variation (str)
+    """
+    def __init__(self,
+            name, variation):
+        Variation.__init__(self, name)
+        self.variation = variation
 
+    def create(self, unit):
+        new_selections = deepcopy(unit.selections)
+        new_actions = deepcopy(unit.actions)
+        if self.variation not in unit.dataset.quantities_per_vars:
+            logger.fatal('Variation {} not found in ntuple'.format(self.variation))
+            raise NameError
+        else:
+            list_of_quants = unit.dataset.quantities_per_vars[self.variation]
+            for quant_i in list_of_quants:
+                for cut_i in range(len(new_selections[0].cuts)):
+                    if quant_i in new_selections[0].cuts[cut_i].expression:
+                        new_selections[0].cuts[cut_i].expression = new_selections[0].cuts[cut_i].expression.replace(quant_i, "{quant}__{var}".format(quant=quant_i, var=self.variation))
+                for weight_i in range(len(new_selections[1].weights)):
+                    if quant_i in new_selections[1].weights[weight_i].expression:
+                        new_selections[1].weights[weight_i].expression = new_selections[1].weights[weight_i].expression.replace(quant_i, "{quant}__{var}".format(quant=quant_i, var=self.variation))
+            for act_i in new_actions:
+                act_i.name = act_i.name.replace("Nominal", self.name)
```

Comment:
I think this is wrong here. The name of the action will already be updated when booking the action object into the unit. You need to change the variable object of the action here, I think.
```suggestion
                act_i.variable = act_i.variable.replace(quant_i, "{quant}__{var}".format(quant=quant_i, var=self.variation))
```

Reply:
I think it would make sense to verify that this function does the shifts you would expect it to do. So for instance for a tau energy scale shift you could produce the histograms of the tau_pt and m_vis and verify that you see a shift in both and check that they look like you would expect.